Bowling/Game.cs

```diff
@@ -0,0 +1,144 @@
+        {
+            if (GameInstance == null)
+            {
+                GameInstance = new Game(numberOfPlayers);
+            }
+            return GameInstance;
+        }
+
+        public void Play()
+        {
+            Random random = new Random();
+            for (var frameIndex = 0; frameIndex < NumberOfFrames; frameIndex++)
+            {
+                foreach (var player in Players)
+                {
+                    player.SimulateRolls();
+                }
+            }
+            foreach (var player in Players)
+                player.SimulateRolls();
+            Calculate();
+        }
+
+        private void GeneratePlayers(int numberOfPlayers)
+        {
+            for (var playerIndex = 0; playerIndex < numberOfPlayers; playerIndex++)
+            {
+                AddPlayer(CreateRandomPlayer());
+            }
+        }
+
+        private IPlayer CreateRandomPlayer()
+        {
+            var random = new Random();
+            var playerType = random.Next(Constants.NumberOfStrategies);
+            switch (playerType)
+            {
+                case 0:
+                    var NormalStrat = new NormalStrategy();
+                    var NormalPlayer = new Player(NameGenerator.Get());
+                    NormalPlayer.SetStrategy(NormalStrat);
+                    return NormalPlayer;
+                default:
+                    var ProStrat = new ProStrategy();
+                    var ProPlayer = new Player(NameGenerator.Get());
+                    ProPlayer.SetStrategy(ProStrat);
+                    return ProPlayer;
+            }
+        }
+
+        private void AddPlayer(IPlayer player)
+        {
+            if (NumberOfPlayers == Players.Count)
+            {
+                throw new AddedMorePlayersThanDeclaredException(Constants.DeclaredAnotherNumberOfPlayers);
+            }
+            else
+            {
+                Players.Add(player);
+            }
+        }
+
+        public void ShowPlayers()
+        {
+            foreach (Player player in Players)
+            {
+                Console.WriteLine(player.Name);
+            }
+        }
+
+        public void DisplayScoreBoard()
+        {
+            Console.ForegroundColor = ConsoleColor.Green;
+            Console.WriteLine(Constants.BowlingBanner);
+            Console.ResetColor();
+            foreach (var player in Players)
+            {
+                Console.ForegroundColor = ConsoleColor.Green;
+                Console.WriteLine(String.Format(Constants.PlayerDisplayFormat, player.Name, player.StrategyName));
+                Console.ResetColor();
+                player.DisplayScore();
+            }
+            Console.ForegroundColor = ConsoleColor.Green;
+            Console.WriteLine(String.Format(Constants.WinnerDisplayFormat, WinnerName, MaxScore));
+            Console.ResetColor();
+        }
+
+        private void Calculate()
```

Comment:
Are we sure that we want this logic here? What do you think if we encapsulate it in that method `Play()` in the for where we loop all the players and we make them play. There inside that class we should make the necessary changes related to the points that a player has. 
The Game class should only care to see at the end who is the winner. This can be done if we loop the List of players and get the one that has the highest score. Maybe this can be done with some `LINQ` methods.
Also if you do like this you will no longer need the WinnerName variable to be here, you will get it with the above logic, because you already have the Name variable inside the Player class